pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_constraint.go

```diff
@@ -11,16 +11,53 @@
 package scbuildstmt
 
 import (
+	"bytes"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/schemaexpr"
+	"github.com/cockroachdb/cockroach/pkg/sql/parser"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgcode"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
+	"github.com/cockroachdb/cockroach/pkg/sql/privilege"
 	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scerrors"
 	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/catid"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/volatility"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/errors"
 )
 
 func alterTableAddConstraint(
 	b BuildCtx, tn *tree.TableName, tbl *scpb.Table, t *tree.AlterTableAddConstraint,
 ) {
-	// Extra checks before reaching this point ensured that the command is
-	// an ALTER TABLE ... ADD PRIMARY KEY command.
+	switch d := t.ConstraintDef.(type) {
+	case *tree.UniqueConstraintTableDef:
+		if !d.PrimaryKey || t.ValidationBehavior != tree.ValidationDefault {
+			panicBecauseGuardRailsIsViolated()
+		}
+		alterTableAddPrimaryKey(b, tn, tbl, t)
+	case *tree.CheckConstraintTableDef:
+		if t.ValidationBehavior != tree.ValidationDefault {
+			panicBecauseGuardRailsIsViolated()
+		}
+		alterTableAddConstraintCheck(b, tn, tbl, t)
+	default:
+		panicBecauseGuardRailsIsViolated()
+	}
+}
+
+// panicBecauseGuardRailsIsViolated is used when previously passed guardrail, `extraChecks`, is now violated.
+func panicBecauseGuardRailsIsViolated() {
+	panic(errors.AssertionFailedf("previously passed guardrail `extraChecks` for ALTER TABLE is now violated."))
+}
```

Comment:
Is this really helpful?

Reply:
it's just that we will be typing this long `panic(......)` line multiple times at its call-sites. Do you have any suggestions for this situation?

Reply:
My point was more that we already check for this in `alter_table.go` and that checking a second time isn't helpful.